### ISSUE TITLE
websocat: WebSocketError/unknown port

### ISSUE THREAD

**Issue by dhruvbalwada** [yuvipanda/jupyter-sshd-proxy#1]
@jbusecke and I have been trying to setup jupyter-sshd-proxy on the LEAP hub. 

Following the instructions in the readme, we setup this image:
Repo for new image: https://github.com/leap-stc/test-image-balwada 
Image: https://quay.io/repository/leap-stc/test-image-balwada?tab=tags&tag=5b146d660ff2 

However, when trying to ssh to this, we get the error:
```
websocat: WebSocketError: WebSocketError: Redirected (302 Found) to /user/dhruvbalwada/sshd
websocat: error running
Connection closed by UNKNOWN port 65535
```

To test if there were any problems with our ssh connections, we tested by sshing to hub with test image (`quay.io/yuvipanda/pangeo-jupyter-sshd-proxy:latest`). This worked and we were able to use terminal to navigate. 

Wondering what we might be doing wrong with our own test image?

**Comment by yuvipanda** [comment]
OOOO sorry for not seeing this before, @dhruvbalwada 

can you post full sshd config, ssh command line used?

**Comment by dhruvbalwada** [comment]
Here is the ssh command line:
```
(base) ➜  ~ ssh LEAP-HUB   
websocat: WebSocketError: WebSocketError: Redirected (302 Found) to /user/dhruvbalwada/sshd
websocat: error running
Connection closed by UNKNOWN port 65535
```

I sent the sshd config on slack, as I was unsure if it is safe to post it here. Let me know what else you need, or if I didn't send the right things.

**Comment by yuvipanda** [comment]
I can reproduce it with this image, but I also don't see `jupyter-sshd-proxy` installed in the image. If I start a server with that image, and do `pip list | grep ssh` I see nothing. I did a `pip list` and also found that the sshd-proxy was not there.

Can you try rebuilding the image and see if `jupyter-sshd-proxy` gets actually installed?

**Comment by jbusecke** [comment]
Hey folks, so I just looked at the [build log]() and it seems like its getting installed (see [here](https://github.com/leap-stc/test-image-balwada/actions/runs/10567214239/job/29275926072#step:5:709) and [here](https://github.com/leap-stc/test-image-balwada/actions/runs/10567214239/job/29275926072#step:5:896)) and then pushed to `quay.io/leap-stc/test-image-balwada:5b146d660ff2`. 

I just tested pulling the image and it is not installed! I faintly remember that I had issues with pip installs following this pattern (there must have been a reason I installed them seperately e.g. [here](https://github.com/jbusecke/pangeo_pyvista_docker_image/blob/698eff3ebb2697d35c57640191931a861ae60d38/Dockerfile#L20) but I cant quite recall). 
Is maybe [`CONDA_DIR`](https://github.com/leap-stc/test-image-balwada/blob/5b146d660ff23de29bdc9fa9464fa537f4ffad10/Dockerfile#L6) not set properly?

**Comment by jbusecke** [comment]
I did the same thing [here](https://github.com/ocean-transport/scale-aware-air-sea/blob/main/Dockerfile) and that worked!

**Comment by jbusecke** [comment]
Or do we need to provide `NB_PYTHON_PREFIX` instead of `CONDA_DIR`? I have to admit I still do not understand [this part of the pangeo docker image magic](https://github.com/pangeo-data/pangeo-docker-images/blob/master/base-image/Dockerfile) very well, which is probably at the root of our problem here?

**Comment by yuvipanda** [comment]
```
FROM pangeo/pangeo-notebook:2024.07.30

RUN mamba install -c conda-forge openssh
RUN pip install --no-cache jupyter-sshd-proxy
```

This is my dockerfile, fwiw

**Comment by jbusecke** [comment]
I think the crux here is that we are using the pangeo-notebook image (which installs into the `notebook` env). I have convinced myself [here and below](https://github.com/leap-stc/test-image-balwada/actions/runs/11075590594/job/30776877820?pr=4#step:5:1010) that we are installing stuff into the base conda env, but we really want it in the notebook one?

Let me test my hypothesis and report back.

**Comment by jbusecke** [comment]
Yep (i believe) I fixed it in https://github.com/leap-stc/test-image-balwada/pull/4. I can now pull `quay.io/leap-stc/test-image-balwada:51661c98ddc8` onto our hub and `pip list |grep ssh` in the terminal shows:
```
jupyter-sshd-proxy            0.2.0
```
@dhruvbalwada could you test with that again?

**Comment by dhruvbalwada** [comment]
It works now! Thank you both for sorting this out.

**Comment by jbusecke** [comment]
I have moved ahead and made a [PR](https://github.com/pangeo-data/pangeo-docker-images/pull/585?notification_referrer_id=NT_kwDOANpsf7QxMjY2NDY0NDkwOToxNDMxNDYyMw&notifications_query=is%3Aunread%20pre-commit#event-14575680510) to the pangeo docker images to get this upstream. Closing this now.

**Comment by yuvipanda** [comment]
I just dug into this a little, turns out that `sshd` is not present in the image. I opened https://github.com/pangeo-data/pangeo-docker-images/pull/588 which I *think* should fix this.

In addition, unless you have aliases set up otherwise, your config should be `Host leap.2i2c.cloud` rather than `Host leap-hub`, as the name there is used in the `%h` in `ProxyCommand`

**Comment by dhruvbalwada** [comment]
I just checked and am still getting the same error, but it also seems that is because leap.2i2c.cloud is not inheriting the latest docker image. Is that something that will happen automatically after some period?

**Comment by jbusecke** [comment]
@dhruvbalwada can you manually check with the latest image version `[2024.10.26](https://github.com/pangeo-data/pangeo-docker-images/releases/tag/2024.10.26)`? 

> Is that something that will happen automatically after some period?

Nope its manual, happy to submit the update once we confirmed that it works.

**Comment by dhruvbalwada** [comment]
What will be image name that I put in when starting the hub? The text in gray above returns invalid image.

**Comment by jbusecke** [comment]
Oh sorry I meant to just just past the tag. You have to enter `quay.io/pangeo/pangeo-notebook:2014.10.16` (as described [here](https://leap-stc.github.io/reference/infrastructure.html#custom-images))

**Comment by dhruvbalwada** [comment]
The tag should be `quay.io/pangeo/pangeo-notebook:2024.10.26`

**Comment by jbusecke** [comment]
Nice! Ill get the default image changed rn.

**Comment by jbusecke** [comment]
https://github.com/2i2c-org/infrastructure/pull/5011
